src/D2L.Security.OAuth2/Utilities/D2LHttpClient.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace D2L.Security.OAuth2.Utilities
+{
+	public class D2LHttpClient : ID2LHttpClient
+	{
+		/// <summary>
+		/// This class implements sync versions of HTTPClient methods.
+		/// TODO: Replace the sync methods which are currently using Task.Run(async verison); task.Wait();
+		/// With (probably) HttpWebRequest
+		/// </summary>
+		private readonly HttpClient m_httpClient;
+
+		public D2LHttpClient() {
+			m_httpClient = new HttpClient();
+		}
+		public D2LHttpClient(HttpMessageHandler handler) {
+			m_httpClient = new HttpClient(handler);
+		}
+		public D2LHttpClient(HttpMessageHandler handler, bool disposeHandler) { 
+			m_httpClient = new HttpClient(handler, disposeHandler);
+		}
+
+		public Task<HttpResponseMessage> GetAsync(string requestUri, HttpCompletionOption completionOption, CancellationToken cancellationToken)
+			=> m_httpClient.GetAsync(requestUri, completionOption, cancellationToken);
+		public Task<HttpResponseMessage> GetAsync(Uri requestUri, CancellationToken cancellationToken)
+			=> m_httpClient.GetAsync(requestUri, cancellationToken);
+		public Task<HttpResponseMessage> GetAsync(Uri requestUri, HttpCompletionOption completionOption, CancellationToken cancellationToken)
+			=> m_httpClient.GetAsync(requestUri, completionOption, cancellationToken);
+		public Task<HttpResponseMessage> GetAsync(Uri requestUri, HttpCompletionOption completionOption)
+			=> m_httpClient.GetAsync(requestUri, completionOption);
+		public Task<HttpResponseMessage> GetAsync(string requestUri, HttpCompletionOption completionOption)
+			=> m_httpClient.GetAsync(requestUri, completionOption);
+		public Task<HttpResponseMessage> GetAsync(Uri requestUri)
+			=> m_httpClient.GetAsync(requestUri);
+		public Task<HttpResponseMessage> GetAsync(string requestUri)
+			=> m_httpClient.GetAsync(requestUri);
+		public Task<HttpResponseMessage> GetAsync(string requestUri, CancellationToken cancellationToken)
+			=> m_httpClient.GetAsync(requestUri, cancellationToken);
+
+		public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
+			=> m_httpClient.SendAsync(request);
+
+		public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+			=> m_httpClient.SendAsync(request, cancellationToken);
+
+		public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption)
+			=> m_httpClient.SendAsync(request, completionOption);
+
+		public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
+			=> m_httpClient.SendAsync(request, completionOption, cancellationToken);
+
+#pragma warning disable D2L0018 // Avoid using dangerous methods

```

Comment:
Would be great if we could use `[DangerousMethods.Audited( ... )] instead of disabling the `#pragma`

Reply:
Is this an existing decorator or are you proposing its creation

Reply:
Existing, just typo'd it:

https://search.d2l.dev/search?project=lms&full=DangerousMethodUsage